packages/react-dom/src/__tests__/ReactDOMFizzServer-test.js

```diff
@@ -10331,4 +10331,49 @@ describe('ReactDOMFizzServer', () => {
       </html>,
     );
   });
+
+  it('can render styles with nonce', async () => {
+    CSPnonce = 'R4nd0m';
+    await act(() => {
+      const {pipe} = renderToPipeableStream(
+        <>
+          <style
+            href="foo"
+            precedence="default"
+            nonce={CSPnonce}>{`.foo { color: hotpink; }`}</style>
+          <style
+            href="bar"
+            precedence="default"
+            nonce={CSPnonce}>{`.bar { background-color: blue; }`}</style>
+        </>,
+      );
+      pipe(writable);
+    });
+    expect(document.querySelector('style').nonce).toBe(
+      CSPnonce,
+    );
+  });
+
+  // @gate __DEV__
```

Comment:
the right way to assert warnings is to use `assertConsoleErrorDev` inside a test that isn't gated specifically on dev environment.

In this case you might want to assert the existed behavior that both rules are included in the final output style tag regardless of the bad nonce and that in dev you get a warning for it. In this case you're not actually asserting that the program output any styles at all so while I suspect it works it'd be good to encode the expected semantic even when the test is about the warning